lunatrace/bsl/frontend/src/components/navbar/Navbar.tsx

```diff
@@ -56,6 +58,11 @@ const NavbarComponent: React.FunctionComponent<NavbarComponentProps> = ({ setupW
     </span>
   );
 
+  const stopImpersonating = () => {
+    setImpersonatedUser(null);
+    window.location.reload();
```

Comment:
This really isnt nuts but doing the context/provider thing I mentioned might eliminate this altogether.  Not saying a reload is a bad idea.